src/vsr/message_header.zig

```diff
@@ -0,0 +1,1430 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+const constants = @import("../constants.zig");
+const stdx = @import("../stdx.zig");
+const vsr = @import("../vsr.zig");
+const Command = vsr.Command;
+const Operation = vsr.Operation;
+const schema = @import("../lsm/schema.zig");
+
+/// Aegis128 (used by checksum) uses hardware accelerated AES via inline asm which isn't
+/// available at comptime. Use a hard-coded value instead and verify via a test.
+const checksum_body_empty: u128 = 0x49F174618255402DE6E7E3C40D60CC83;
+
+test "empty checksum" {
+    assert(checksum_body_empty == vsr.checksum(&.{}));
+}
+
+/// Network message and journal entry header:
+/// We reuse the same header for both so that prepare messages from the primary can simply be
+/// journalled as is by the backups without requiring any further modification.
+pub const Header = extern struct {
```

Comment:
Currently planned changes to frame fields:

- Change all checksums to `u256`. (For future AEAD tags.)
- Change nonce to `u256`. (For future AEAD nonce.)
- Change `cluster` to `u128`. (I still don't understand the reasoning here, but we have room so why not. :shrug:)

Reply:
Great!

A 128-bit `cluster` also improves our "resolution" for interoperability with other systems that want to use UUIDs for cluster ID to allow for random generation of cluster ID without the cost of having to introduce a centralized ID coordinator. Plus it gives us a further security guarantee that clusters can't talk to other clusters unless they can literally “guess” the identifier, which is now cryptographically "tough" (granted, we might not be able to imagine now why this could be important, but it's a stronger property nevertheless—especially from an online verification point of view).

(The motivation for AEADs moving to nonces bigger than 96-bit... is also why 128-bit is a “bare minimum” for cluster ID for us).